onnx/defs/schema.h

```diff
@@ -1184,16 +1184,23 @@ class OpSchemaRegistry final : public ISchemaRegistry {
     // standard ONNX domains as above). Custom-domains are free to interpret
     // this as appropriate (that is, as relative to releases of custom-domain
     // as opposed to ONNX releases).
-    void
-    AddDomainToVersion(const std::string& domain, int min_version, int max_version, int last_release_version = -1) {
+    void AddDomainToVersion(
+        const std::string& domain,
+        int min_version,
+        int max_version,
+        int last_release_version = -1,
+        bool rewrite = false) {
       std::lock_guard<std::mutex> lock(mutex_);
-      assert(map_.end() == map_.find(domain));
+      if (!rewrite) {
+        // Maybe we need raise an exception when domain is already exist.
+        assert(map_.end() == map_.find(domain));
+        assert(last_release_version_map_.end() == last_release_version_map_.find(domain));
+      }
```

Comment:
@gramalingam for an opinion

Reply:
Why is this needed? Not clear to me. (If it is required, may be a separate `Update` method would be better than adding the `rewrite` parameter.)

Reply:
Sure, I will implement the `Update` method instead of the `rewrite` flag. And should we considering to replacing the `assert` with a fatal exception to ensure consistent behavior between debug and release modes.